CONTRIBUTING.md

```diff
@@ -0,0 +1,129 @@
+
+This way your forked or local copy of Vugu can correspond to the import path `github.com/vugu/vugu`.  Example:
+
+```bash
+git checkout https://github.com/vugu/vugu # or, your forked copy
+```
+
+And now you can reference it from another project:
+
+some-other-project/go.mod:
+```
+module some-other-project
+
+// point vugu imports at your local copy instead of public download
+replace github.com/vugu/vugu => ../vugu
+```
+
+## Vugu Project Layout
+
+Some important subdirectories in the Vugu project are:
+* The top level (corresponding to `github.com/vugu/vugu` import path) has common things shared across the other packages.
+* `cmd` contains command-line executables, `vugugen` being the most commonly used.  Note that these command-line tools are generally small and just a thin wrapper around functionality from another package.
+* `devutil` has utilities specifically created to facility rapid development such as wrappers for each Go+Wasm compilation and an HTTP muxer more suitable for Vugu development than the default one from net/http.
+* `domrender` has the client-side code which synchronizes Vugu's virtual DOM tree with the browser's DOM.
+* `gen` has code generaiton logic used to convert .vugu files into .go source code.  Most of the implementation of `vugugen` is available here.
+* `js` has a wrapper around `syscall/js` which delegates to the default implementation in WebAssembly but in other environments allows compilation and functionality gracefully degrades.  Useful for writing code that needs to compile both in and outside Wasm environments without having to maintain two separate implementations via build tags just to include one reference to e.g. js.Value.
+* The following folders are becoming deprecated and should be used with caution: `simplehttp` is an earlier attempt at development tooling use `devutil` instead.  `tinygo-dev` will be removed, the working parts have moved into `devutil`.
+* `wasm-test-suite` contains a test suite using Headless Chrome, see Running Tests below.
+
+## Running Tests
+
+In `wasm-test-suite`, `run-wasm-test-suite-docker.sh` can be used to launch a docker container with Headless Chrome and a small server.  You can then use `go test` in `wasm-test-suite` to execute the various end-to-end tests available.
+
+This requires Docker to be installed properly and the test suite expects to be able to connect the running docker instance on ports 8846 for web serving/upload and port 9222 to control Headless Chrome.
+
+## Vugu Documentation
+
+Documention for Vugu, other than GoDoc (https://pkg.go.dev/github.com/vugu/vugu), lives on https://vugu.org.  The source for it is at https://github.com/vugu/vugu-site and you can submit PRs to that repository to propose changes.  For small improvements and errata it is fine to just submit a PR.  For more significant changes or if it seems like discussion is require, please make a GitHub issue first.
```

Comment:
Instead of "For more significant changes or if it seems like discussion is require, please make a GitHub issue first" you could "For more significant changes or if a discussion is required, please create a GitHub issue first"